docs/Development.md

```diff
@@ -0,0 +1,59 @@
+
+
+### Get the source code
+Besides the omsagent core itself, OMS Agent consists of a number of other components and their kits (shell bundles).
+We have a root project called [Build-OMS-Agent-for-Linux](https://github.com/Microsoft/Build-OMS-Agent-for-Linux) to build them all; it references the 10 components of the agent as submodules.
+
+#### Submodules
+- auoms/auoms-kits (Auditd plugin for OMS agent) 
+- dsc/dsc-kits (Desired State Configuration) 
+- omi/omi-kits (Open Management Infrastructure) 
+- opsmgr-kits (Operations Manager) 
+- pal (Platform Abstraction Layer) 
+- scxcore-kits (System Center Cross Platform Provider for Operations Manager) 
+
+To build the bundle, you will need to clone the Build-OMS-Agent-for-Linux repository.  
+```
+git clone --recursive git@github.com:Microsoft/Build-OMS-Agent-for-Linux.git /home/$(whoami)
+
+cd /home/$(whoami)/Build-OMS-Agent-for-Linux/omsagent
+git checkout master
+git checkout -b my-feature
+```
+
+### Set up environment with a build container (Docker)
+Our build environment is based on Centos, you can setup your own dev container
+by choosing the right docker file under the ./build/docker folder.
+
+Example using Dockerfile.centos6:
+```
+docker build -t oms-centos6-x64 -f /home/$(whoami)/Build-OMS-Agent-for-Linux/omsagent/build/docker/Dockerfile.centos6 .
+```
+
+You can instead pull the existing container image from Docker Hub repository. You should pull this image often to keep your local build up-to-date:
+```
+docker pull abenbachir/oms-centos6-x64
+```
+
+Then you can start a build container instance:
+```
+docker run --rm -itv /home/$(whoami)/:/home/scratch/Build-OMS-Agent-for-Linux -w /home/scratch/Build-OMS-Agent-for-Linux/omsagent/build abenbachir/oms-centos6-x64:latest
+```
+
+### Building steps
+Once inside the build container, do the following to build omsagent:
+```
+./configure --enable-ulinux
```

Comment:
no jemalloc flag?

Reply:
I'm waiting until v1.11 is out, if there is no issue with jemalloc I will enabled it by default.